File: src/main/js/app/nuxeo/rpc/query.js
/*
 * (C) Copyright 2014 Nuxeo SA (http://nuxeo.com/) and contributors.
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the GNU Lesser General Public License
 * (LGPL) version 2.1 which accompanies this distribution, and is available at
 * http://www.gnu.org/licenses/lgpl-2.1.html
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 *
 * Contributors:
 *     Nelson Silva <nelson.silva@inevo.pt>
 */
import {Operation} from './operation';

/**
 * Paginated Query
 */
class Query extends Operation {

  constructor(connection) {
    super(connection, 'Document.PageProvider');
  }

  set nxql(q) {
    this.params.language = 'nxql';
    this.params.query = q;
  }

  set namedParameters(p) {
    this.params.namedParameters = p;
  }

  set queryParameters(p) {
    this.params.queryParams = p;
  }

  set pageProvider(name) {
    this.params.providerName = name;
  }

  get page() {
    return this.params.currentPageIndex;
  }

  set page(p) {
    this.params.currentPageIndex = p;
  }

  set pageSize(s) {
    this.params.pageSize = s;
  }

  set sortBy(s) {
    this.params.sortBy = s.join(',');
  }

  set sortOrder(s) {
    this.params.sortOrder = s.join(',');
  }

  run() {
    return this.execute();
  }
}

export {Query};

File: nuxeo-features/nuxeo-automation/nuxeo-automation-shell/src/main/java/org/nuxeo/ecm/shell/automation/cmds/Query.java
/*
 * (C) Copyright 2006-2010 Nuxeo SAS (http://nuxeo.com/) and contributors.
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the GNU Lesser General Public License
 * (LGPL) version 2.1 which accompanies this distribution, and is available at
 * http://www.gnu.org/licenses/lgpl.html
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 *
 * Contributors:
 *     bstefanescu
 */
package org.nuxeo.ecm.shell.automation.cmds;

import org.nuxeo.ecm.automation.client.jaxrs.model.Document;
import org.nuxeo.ecm.automation.client.jaxrs.model.Documents;
import org.nuxeo.ecm.shell.Argument;
import org.nuxeo.ecm.shell.Command;
import org.nuxeo.ecm.shell.Context;
import org.nuxeo.ecm.shell.Parameter;
import org.nuxeo.ecm.shell.ShellConsole;
import org.nuxeo.ecm.shell.ShellException;
import org.nuxeo.ecm.shell.automation.DocRefCompletor;
import org.nuxeo.ecm.shell.automation.DocumentHelper;
import org.nuxeo.ecm.shell.automation.RemoteContext;

/**
 * @author <a href="mailto:bs@nuxeo.com">Bogdan Stefanescu</a>
 * 
 */
@Command(name = "query", help = "Query documents")
public class Query implements Runnable {

    @Context
    protected RemoteContext ctx;

    @Parameter(name = "-uid", hasValue = false, help = "If used the matching documents will be printed using the document UID.")
    protected boolean uid = false;

    @Argument(name = "query", index = 1, required = false, completor = DocRefCompletor.class, help = "The document path")
    protected String query;

    public void run() {
        try {
            Documents docs = ctx.getDocumentService().query(query);
            ShellConsole console = ctx.getShell().getConsole();
            for (Document doc : docs) {
                if (uid) {
                    console.println(doc.getId());
                } else {
                    DocumentHelper.printPath(console, doc);
                }
            }
        } catch (Exception e) {
            throw new ShellException(e);
        }
    }
}
